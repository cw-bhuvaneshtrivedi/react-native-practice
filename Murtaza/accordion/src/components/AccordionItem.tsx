import { StyleSheet, Text, View, Dimensions } from "react-native";
import React from "react";

interface AccordionItemProps {
  itemName: string;
}
const AccordionItem: React.FC<AccordionItemProps> = ({ itemName }) => {
  return (
    <View style={styles.container} testID="accordion-item">
      <View style={styles.item}>
        <Text style={styles.text} >{itemName}</Text>
      </View>
      <View style={styles.line}></View>
    </View>
  );
};

export default AccordionItem;

const styles = StyleSheet.create({
  container: {
    width: Dimensions.get("screen").width,
  },
  item: {
    backgroundColor: "#f9f9f9",
    paddingLeft: 36,
    paddingTop: 16,
    paddingBottom: 16,
    height: 50,
  },
  text: {
    fontSize: 15,
    color: "#484848",
  },
  line: {
    backgroundColor: "#efefef",
    width: "100%",
    height: 1,
  },
});
