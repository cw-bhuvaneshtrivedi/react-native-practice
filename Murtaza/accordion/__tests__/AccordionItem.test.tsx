import React from 'react';
import { render } from '@testing-library/react-native';
import AccordionItem from '../src/components/AccordionItem';

test('renders AccordionItem correctly', () => {
  const { toJSON } = render(<AccordionItem />);
  expect(toJSON()).toMatchSnapshot();
});
