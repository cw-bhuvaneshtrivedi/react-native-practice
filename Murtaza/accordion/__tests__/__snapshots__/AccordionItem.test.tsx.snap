// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AccordionItem correctly 1`] = `
<View
  style={
    {
      "width": 750,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#f9f9f9",
        "height": 50,
        "paddingBottom": 16,
        "paddingLeft": 36,
        "paddingTop": 16,
      }
    }
  >
    <Text
      style={
        {
          "color": "#484848",
          "fontSize": 15,
        }
      }
    >
      AccordionItem
    </Text>
  </View>
  <View
    style={
      {
        "backgroundColor": "#efefef",
        "height": 1,
        "width": "100%",
      }
    }
  />
</View>
`;
